internal/transport/http2_client.go

```diff
@@ -201,6 +201,12 @@ func newHTTP2Client(connectCtx, ctx context.Context, addr resolver.Address, opts
 		}
 	}()
 
+	// gRPC, resolver, balancer etc. can specify arbitrary data in the
+	// Attributes field of resolver.Address, which is shoved into connectCtx
+	// and passed to the dialer and credential handshaker. This makes it possible for
+	// address specific arbitrary data to reach custom dialers and credential handshakers.
+	connectCtx = icredentials.NewClientHandshakeInfoContext(connectCtx, credentials.ClientHandshakeInfo{Attributes: addr.Attributes})
```

Comment:
I understand the purpose, but I don't see how it works.
This information can only be retrieved using the _internal_ creds package. How would a dialer make use of it? Or are we going to change the default dialer?

Another concern is the name. This is the "handshake" info, not dial info.